src/client.js

```diff
@@ -72,12 +72,12 @@ const Client = class Client {
 
   _handleMessage(topic, message) {
     var parts = topic.split('/');
-    if (parts[3] !== 'activations' && parts[3] !== 'up') {
+    if (parts[4] !== 'activations' && parts[3] !== 'up') {
       return;
     }
     var devId = parts[2];
     var payload = JSON.parse(message.toString());
-    if (parts.length === 4) { // inject app_id and dev_id, but not for field topics
+    if (parts[3] !== 'up' || parts.length === 4) { // inject app_id and dev_id, but not for field topics
       payload.app_id = this.appId;
       payload.dev_id = devId;
     }
```

Comment:
I am against putting `appId` and `devId` in the payload. We are already providing the `devId` in the callback, so that is already convenient for the developer. The `devId` is not part of the _payload_, and should imo not be added by a SDK. Let's make sure that payload is exactly what is sent by the device, possibly decoded and converted.